tmt/steps/provision/testcloud.py

```diff
@@ -531,6 +541,79 @@ def _generate_disk_filepaths() -> Iterator[Path]:
             )
 
 
+def _apply_hw_cpu_processors(
+        hardware: Optional[tmt.hardware.Hardware],
+        domain: 'DomainConfiguration',
+        logger: tmt.log.Logger) -> None:
+    """ Apply ``cpu.processors`` constraint to given VM domain """
+
+    domain.cpu_count = DEFAULT_CPU_COUNT
+
+    if not hardware or not hardware.constraint:
```

Comment:
@happz I am wondering, should this general check be somewhere before this function is even called?